src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/service/NetworkIslService.java

```diff
@@ -171,61 +169,46 @@ public void bfdPropertiesUpdate(IslReference reference) {
 
     /**
      * Process installed isl rule notification.
-     * @param reference isl reference
-     * @param payload response payload
      */
-    public void islDefaultRuleInstalled(IslReference reference, InstallIslDefaultRulesResult payload) {
+    public void islDefaultRuleInstalled(IslReference reference, Endpoint endpoint) {
         log.debug("ISL service received isl rule installed for {} (on {})", reference, reference.getSource());
         IslFsm islFsm = locateController(reference);
         IslFsmContext context;
-        if (payload.isSuccess()) {
-            context = IslFsmContext.builder(carrier, reference.getSource())
-                    .installedRulesEndpoint(Endpoint.of(payload.getSrcSwitch(), payload.getSrcPort()))
-                    .build();
-            controllerExecutor.fire(islFsm, IslFsmEvent.ISL_RULE_INSTALLED, context);
-        } else {
-            context = IslFsmContext.builder(carrier, reference.getSource())
-                    .build();
-            controllerExecutor.fire(islFsm, IslFsmEvent.ISL_RULE_TIMEOUT, context);
-        }
+        context = IslFsmContext.builder(carrier, reference.getSource())
+                .endpoint(endpoint)
+                .build();
+        controllerExecutor.fire(islFsm, IslFsmEvent.ISL_RULE_INSTALLED, context);
     }
 
     /**
      * Process removed isl rule notification.
-     * @param reference isl reference
-     * @param payload target endpoint
      */
-    public void islDefaultRuleDeleted(IslReference reference, RemoveIslDefaultRulesResult payload) {
+    public void islDefaultRuleDeleted(IslReference reference, Endpoint endpoint) {
         log.debug("ISL service received isl rule removed for {} (on {})", reference, reference.getSource());
         IslFsm controller = this.controller.get(reference);
         if (controller == null) {
             log.info("Got clean up resources notification for not existing ISL {}", reference);
             return;
         }
         IslFsmContext context;
-        if (payload.isSuccess()) {
-            context = IslFsmContext.builder(carrier, reference.getSource())
-                    .removedRulesEndpoint(Endpoint.of(payload.getSrcSwitch(), payload.getSrcPort()))
-                    .build();
-            controllerExecutor.fire(controller, IslFsmEvent.ISL_RULE_REMOVED, context);
-        } else {
-            context = IslFsmContext.builder(carrier, reference.getSource()).build();
-            controllerExecutor.fire(controller, IslFsmEvent.ISL_RULE_TIMEOUT, context);
-        }
+        context = IslFsmContext.builder(carrier, reference.getSource())
+                .endpoint(endpoint)
+                .build();
+        controllerExecutor.fire(controller, IslFsmEvent.ISL_RULE_REMOVED, context);
 
         removeIfCompleted(reference, controller);
```

Comment:
Why there is no such call into `islDefaultRuleInstalled` method?

PS installed and removed methods are the same, the only used event is different.

Reply:
Merged two methods into one.